service SkillitorCore {
    rpc SetSkill (SkillAssociation) returns Acknowledgement
    rpc UnsetSkill (SkillAssociation) returns Acknowledgement
    rpc FindSkill (SkillSpec) returns SkillAssociation
}

message SkillAssociation {
    string user_id = 1;
    repeated Skillspec skill = 2;
}

message SkillSpec {
    string skill_name = 1;
    enum skill_level {
        UNDEFINED = 0;
        BEGINNER = 1;
        INTERMEDIATE = 2;
        ADVANCED = 3;
    }
}

message Acknowledgement {
    bool success;
    string error_msg;
}
